### ISSUE TITLE
Logging analyzer CA2017 does not work when the params argument is passed as an array

### ISSUE THREAD

**Issue by aqueenan** [dotnet/roslyn-analyzers#6224]
### Analyzer

**Diagnostic ID**: [CA2017](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2017)

### Describe the improvement

When exactly one parameter is supplied and that parameter is an array, the new logging analyzer CA2017 (added in https://github.com/dotnet/runtime/issues/36064) does not detect mismatches between the number of parameters supplied and the number of placeholders in the template string because the number of parameters can vary at runtime.

```cs
ILogger logger = LoggerFactory.Create(
	builder =>
	{
		builder.AddConsole();
	})
	.CreateLogger("Default");

logger.LogInformation("Command line arguments: {args}", Array.Empty<string>());
// Output: Command line arguments: {args}

logger.LogInformation("Command line arguments: {args}", new [] { "arg1" });
// Output: Command line arguments: arg1

logger.LogInformation("Command line arguments: {args}", new [] { "arg1", "arg2" });
// Output: Command line arguments: arg1
```

### Describe suggestions on how to achieve the rule

When exactly one parameter is supplied and that parameter is an array, show the warning and recommend casting the array to an object.

```cs
ILogger logger = LoggerFactory.Create(
	builder =>
	{
		builder.AddConsole();
	})
	.CreateLogger("Default");

logger.LogInformation("Command line arguments: {args}", (object) Array.Empty<string>());
// Output: Command line arguments: []

logger.LogInformation("Command line arguments: {args}", (object) new [] { "arg1" });
// Output: Command line arguments: [ 'arg1' ]

logger.LogInformation("Command line arguments: {args}", (object) new [] { "arg1", "arg2" });
// Output: Command line arguments: [ 'arg1', 'arg2' ]
```

As an alternative, create a new rule:

> CA20xx: Runtime parameter count mismatch
> Number of parameters supplied in the logging message template may not match the number of named placeholders at runtime. To pass an array as a single parameter, cast it to object.
